presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchTableDescription.java

```diff
@@ -0,0 +1,167 @@
+
+/**
+ * Json description to parse a Elasticsearch table.
+ */
+public class ElasticsearchTableDescription
+{
+    private final String tableName;
+    private final String schemaName;
+    private final String hostAddress;
+    private final int port;
+    private final String clusterName;
+    private final String index;
+    private final boolean indexExactMatch;
+    private final String type;
+
+    private List<ElasticsearchColumn> columns;
+    private List<ElasticsearchColumnMetadata> columnsMetadata;
+
+    @JsonCreator
+    public ElasticsearchTableDescription(
+            @JsonProperty("tableName") String tableName,
+            @JsonProperty("schemaName") String schemaName,
+            @JsonProperty("hostAddress") String hostAddress,
+            @JsonProperty("port") int port,
+            @JsonProperty("clusterName") String clusterName,
+            @JsonProperty("index") String index,
+            @JsonProperty("indexExactMatch") boolean indexExactMatch,
+            @JsonProperty("type") String type,
+            @JsonProperty("columns") List<ElasticsearchColumn> columns)
+    {
+        checkArgument(!isNullOrEmpty(tableName), "tableName is null or is empty");
+        checkArgument(!isNullOrEmpty(hostAddress), "hostAddress is null or is empty");
+        checkArgument(!isNullOrEmpty(String.valueOf(port)), "port is null or is empty");
+        checkArgument(!isNullOrEmpty(clusterName), "clusterName is null or is empty");
+        checkArgument(!isNullOrEmpty(type), "type is null or is empty");
+        this.tableName = tableName;
+        this.schemaName = schemaName;
+        this.hostAddress = hostAddress;
+        this.port = port;
+        this.clusterName = clusterName;
+        this.index = index;
+        this.indexExactMatch = indexExactMatch;
+        this.type = type;
+        if (columns == null) {
+            this.columns = null;
+        }
+        else {
+            this.columns = ImmutableList.copyOf(columns);
+        }
+    }
+
+    @JsonProperty
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    @JsonProperty
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @JsonProperty
+    public String getHostAddress()
+    {
+        return hostAddress;
+    }
+
+    @JsonProperty
+    public int getPort()
+    {
+        return port;
+    }
+
+    @JsonProperty
+    public String getClusterName()
+    {
+        return clusterName;
+    }
+
+    @JsonProperty
+    public String getIndex()
+    {
+        return index;
+    }
+
+    @JsonProperty
+    public boolean getIndexExactMatch()
+    {
+        return indexExactMatch;
+    }
+
+    @JsonProperty
+    public String getType()
+    {
+        return type;
+    }
+
+    @JsonProperty
+    public List<ElasticsearchColumn> getColumns()
+    {
+        return columns;
+    }
+
+    public void setColumns(List<ElasticsearchColumn> columns)
+    {
+        this.columns = columns;
```

Comment:
Make a defensive copy (currently the caller makes it).

Reply:
get it. will update